WordPress/Classes/Networking/LoginClient.swift

```diff
@@ -78,4 +81,134 @@ actor LoginClient {
             "error": error.localizedDescription
         ])
     }
+
+    @MainActor
+    static func displaySelfHostedLoginView(in navigationController: UINavigationController) {
+        let loginClient = LoginClient(session: .shared)
+        let vm = SelfHostedLoginViewModel(loginClient: loginClient)
+        let loginViewController = UIHostingController(rootView: LoginWithUrlView(viewModel: vm))
+
+        vm.onLoginComplete = {
+            loginViewController.dismiss(animated: true)
```

Comment:
This might be a retain cycle. It's also unclear why `pushViewController` is followed by the `loginViewController.dismiss` call .

Reply:
Good catch! Fixed in https://github.com/wordpress-mobile/WordPress-iOS/pull/23473/commits/d8e1b1e99fbdfa910637638291c5c60da6a23a7b